### DISCUSSION TITLE
Examples cases with analytical Jacobians

### DISCUSSION THREAD

**Discussion by cuihantao** [NREL-SIIP/PowerSimulationsDynamics.jl#140] [Q&A]
Is there any example cases for obtaining the analytical Jacobians?

I'm interested but was not able to get the SMIB case working. When creating the perturbation, the code in the Quick Start Guide shows an error (also in the docs). 

Thanks,
Hantao

**Comment by rodrigomha** [comment]
Thanks for the catch-up on this. We will fix the Quick Start Guide bug for the next release.

Regarding Jacobians, it is not possible to obtain analytical Jacobians (as a symbolic function I mean). However, and @jd-lara can clarify more on this, it is possible to obtain the Jacobian function that is obtained via Automatic Differentiation, that can be evaluated for any value of the states.

Nevertheless, most of this is used internally, and we currently do not provide an easy way for users to access this function and play with it. @jd-lara, I think it could be valuable to provide some interface to that for users. 

@cuihantao What would you like to see for example if you have access to this Jacobian function?

**Reply by jd-lara** [reply]
@cuihantao we use ForwardDifferentiation to make the exact jacobians but we currently don't support easy direct access to them. It is more of an advanced feature at the moment. 

We need to define a clean interface for it. Since you are more of an advanced user, [here](https://github.com/NREL-SIIP/PowerSimulationsDynamics.jl/blob/master/src/base/jacobian.jl) is the code related to the Jacobian calculations. 

In a nutshell, what we do is use a cache of the dual types that allow getting the FF evaluation and use a cache of the same making to evaluate that FF with floats.

**Comment by cuihantao** [comment]
@rodrigomha Great. Thanks for getting back!

I would like to see the elements in a Jacobian matrix for a sample system. As @jd-lara mentioned, I will look into the code for now.
